src/XMakeCommandLine/XMake.cs

```diff
@@ -226,6 +226,21 @@ private static void HandleConfigurationException(Exception ex)
         }
 
         /// <summary>
+        /// Insert the command executable path as the first element of the args array.
+        /// </summary>
+        /// <param name="args"></param>
+        /// <returns></returns>
+        private static string[] ConstructArrayArg(string[] args)
```

Comment:
Agreed, just don't like having multiple implementations of logic. Could also just always use args[] parameter, but I'm sure there would be some impact in some corner case :).

I'm ok with this if we don't want to change it either way.